cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -336,30 +336,30 @@ std::pair<VersionedItem, FrameAndDescriptor> LocalVersionedEngine::read_datafram
     return std::make_pair(version.value_or(VersionedItem{}), std::move(frame_and_descriptor));
 }
 
-std::pair<VersionedItem, std::optional<google::protobuf::Any>> LocalVersionedEngine::read_descriptor_version_internal(
+std::pair<VersionedItem, std::optional<google::protobuf::Any>> LocalVersionedEngine::read_descriptor_internal(
         const StreamId& stream_id,
         const VersionQuery& version_query
     ) {
     ARCTICDB_SAMPLE(ReadDescriptor, 0)
-
-    auto version = get_version_to_read(stream_id, version_query);
-    missing_data::check<ErrorCode::E_NO_SUCH_VERSION>(static_cast<bool>(version),
+    auto metadata = read_metadata_internal(stream_id, version_query);
```

Comment:
Please use [structured binding](https://en.cppreference.com/w/cpp/language/structured_binding) to assign `.first` and `.second` to properly named variables. Otherwise, this code is quite cryptic.

Reply:
done